### ISSUE TITLE
bundle update <gem> should say when it could not update to latest and why

### ISSUE THREAD

**Issue by grosser** [bundler/bundler#7453]
A few times already I did `bundle update faraday` for example just for it to do nothing then I have to hop over to rubygems to check what the latest is or go through the Gemfile.lock to see who is locking it.

It would be great to either have `bundle update faraday --force-latest` and then fail/explain "Version 1.2.3 could not be installed because <regular dependency dump>" or have the regular `bundle update faraday` say `updated faraday to 1.0.0 (cannot update to 1.2.3 because of foobar dependency)`

**Comment by Drenmi** [comment]
As you pointed out, there are two cases where the Gem would remain at the same version:

1. It is already at the latest version.
2. Its version is locked by some dependent.

I'm going to venture a guess that the first case is easier to detect than doing the reverse lookup, so maybe this feature's implementation could be split into two stages:

**Messages in first implementation:**

1. "Bundler attempted to update faraday but it is already at the latest version"
2. "Bundler attempted to update faraday but its version is locked by another gem"

**Messages in the improved implementation:**

1. "Bundler attempted to update faraday but it is already at the latest version"
2. "Bundler attempted to update faraday but its version is locked by foobar"

**Comment by grosser** [comment]
yeah, "locked by another gem" is good enough ... pretty easy to hunt it
down after that :)
